server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationClusterSettingIT.java

```diff
@@ -123,4 +133,262 @@ public void testIndexReplicationSettingOverridesDocRepClusterSetting() throws Ex
         assertEquals(indicesService.indexService(anotherIndex).getIndexSettings().isSegRepEnabled(), false);
     }
 
+    public void testDifferentReplicationTypes_CreateIndex_StrictMode() throws Throwable {
```

Comment:
Can you validate this logic in the more lightweight [unit test](https://github.com/opensearch-project/OpenSearch/blob/3b313598d413fc50adee4d4d6ed2ef1d95e5b1c0/server/src/test/java/org/opensearch/cluster/metadata/MetadataCreateIndexServiceTests.java) as opposed to defining all these integration test scenarios?

Reply:
Thanks @andrross. Added unit tests and removed few integration tests. I kept few integration tests for basic sanity. Let me know if you want me to remove all integration tests.